.github/workflows/mvn-verify-check.yml

```diff
@@ -140,4 +147,31 @@ jobs:
           -DskipTests
           -Dskip
           -Dmaven.javadoc.skip
-          
\ No newline at end of file
+
+  verify-modules-with-jdk17:
+    needs: get-shim-versions-from-dist
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJSON(needs.get-shim-versions-from-dist.outputs.sparkJDK17Versions) }}
+    steps:
+      - uses: actions/checkout@v3 # refs/pull/:prNumber/merge
+
+      - name: Setup Java and Maven Env
+        uses: actions/setup-java@v3
+        with:
+          distribution: adopt
+          java-version: 17
+
+      - name: Build JDK17
+        run: >
+          mvn -Dmaven.wagon.http.retryHandler.count=3 -B verify
+          -P 'individual,pre-merge'
+          -Dbuildver=${{ matrix.spark-version }}
+          -Dmaven.compiler.release=17
+          -Dmaven.compiler.source=17
+          -Dmaven.compiler.target=17
```

Comment:
Wondering if it would be better to encapsulate all the property changes needed for JDK17 into a jdk17 Maven profile in pom.xml to make it easier to use and maintain.  That profile was already added to dist/pom.xml, so it'd be nice to make it as easy as -Pjdk17 to build for JDK17.

Reply:
sure, I will make an update to the profiles of jdk 11 and 17

Reply:
Keep existing dist profiles there to avoid breaking other usages, and add new profiles <jdk11|jdk17-build> in parent pom for jdk11+ build properties